integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-spark-service-bundle/nifi-spark-service/src/main/java/com/thinkbiganalytics/nifi/v2/spark/SparkJobserverService.java

```diff
@@ -0,0 +1,353 @@
+
+            syncTimeout = context.getProperty(SYNC_TIMEOUT).getValue();
+            setupContextTimeoutTask();
+
+        } catch (Exception ex) {
+            getLogger().error(ex.getMessage());
+        }
+        getLogger().info("Created new connection to Spark Jobserver: " + jobServerUrl);
+    }
+
+    @Override
+    public boolean checkIfContextExists(String contextName) {
+
+        Boolean contextExists = false;
+
+        try {
+            contextExists = client.getContexts().contains(contextName);
+
+            if (!contextExists) {
+                getLogger().info(String.format("Context %s not found on Spark Jobserver %s", contextName, jobServerUrl));
+            } else {
+                getLogger().info(String.format("Context %s exists on Spark Jobserver %s", contextName, jobServerUrl));
+                contextExists = true;
+            }
+
+        } catch (Exception ex) {
+            getLogger().error(ex.getMessage());
+        }
+        return contextExists;
+    }
+
+    @Override
+    public boolean createContext(String contextName, String numExecutors, String memPerNode, String numCPUCores, SparkContextType contextType, int contextTimeout, boolean async) {
+
+        boolean newContext = isNewContext(contextName);
+
+        if (newContext) {
+            Boolean contextRunning = checkIfContextExists(contextName);
+            if (!contextRunning) {
+                try {
+                    Map<String, String> params = new HashMap<>();
+                    params.put(ISparkJobServerClientConstants.PARAM_NUM_EXECUTORS, numExecutors);
+                    params.put(ISparkJobServerClientConstants.PARAM_SPARK_EXECUTOR_MEMORY, memPerNode);
+                    params.put(ISparkJobServerClientConstants.PARAM_NUM_CPU_CORES, numCPUCores);
+
+                    if (contextType != SparkContextType.SPARK_CONTEXT) {
+                        if (contextType == SparkContextType.SQL_CONTEXT) {
+                            params.put(ISparkJobServerClientConstants.PARAM_CONTEXT_TYPE, "spark.jobserver.context.SQLContextFactory");
+                        } else if (contextType == SparkContextType.HIVE_CONTEXT) {
+                            params.put(ISparkJobServerClientConstants.PARAM_CONTEXT_TYPE, "spark.jobserver.context.HiveContextFactory");
+                        }
+                    }
+
+                    if (async) {
+                        params.put(ISparkJobServerClientConstants.PARAM_SYNC, "false");
+                    } else {
+                        params.put(ISparkJobServerClientConstants.PARAM_SYNC, "true");
+                        params.put(ISparkJobServerClientConstants.PARAM_TIMEOUT, syncTimeout);
+                    }
+
+                    getLogger().info(
+                        String.format("Creating %s %s with %s executors, %s cores and %s memory per executor on Spark Jobserver %s", contextType, contextName, numExecutors, numCPUCores, memPerNode,
```

Comment:
Please allow `info` to format the message rather than using `String.format`. This is more performant if the INFO logging is not enabled.